### PR TITLE
Add other models

### PR DESCRIPTION
**Pull request by JamesHWade**

### CODE REVIEW

**Comment by JamesHWade** [comment]
@calderonsamuel here is what I have so far. It's not working yet, but I hope to have something workable in a few days.

**Comment by calderonsamuel** [comment]
I see. At first glance, I have two comments:

1. I understand this is not inmediately clear, but there is no need to have different UIs for stream and no-stream. If we don't pass any arguments to `StreamHandler$new()` the app will just wait for the full response and rendered it once it arrives, behaving exactly as if the stream is not handled at all. `renderStreamingMessage()` is only displayed when a `StreamHandler()` is initialized with both the shiny session and the user prompt.
2. I agree that having S3 methods is a good step for extendability infrastructure, however we should agree on some naming conventions (`request_base_hf()`, `query_huggingface_api ()`, `hf_create_completion()` are all different!). I was even thinking on something like `base_request(api = "huggingface")` with classes `c("huggingface", "llmrequest", "httr2_request")` or something similar.

File: R/mod_chat.R [hunk 1]
```diff
@@ -46,38 +46,36 @@ mod_chat_server <- function(id, ide_colors = get_ide_theme_info()) {
...
       prompt$chat_history %>%
         style_chat_history(ide_colors = ide_colors)
     }) %>%
       bindEvent(prompt$chat_history, prompt$clear_history)
 
-
     shiny::observe({
       stream_handler <- StreamHandler$new(
         session = session,
         user_prompt = prompt$input_prompt
```
Comment (calderonsamuel) [inline R/mod_chat.R:64]: Removing these two lines removes rendering of the stream

**Comment by JamesHWade** [comment]
Thank you for clarifying. I clearly need to read your blog post in more detail 😊. 

Standardizing our naming convention is a must. Good idea! Maybe a good discussion thread topic?

**Comment by calderonsamuel** [comment]
> Thank you for clarifying. I clearly need to read your blog post in more detail 😊.
> 
> Standardizing our naming convention is a must. Good idea! Maybe a good discussion thread topic?

Great! I've been thinking about this for the past few days. I'll open a thread about this by the end of the day.

**Comment by calderonsamuel** [comment]
Sorry for the delay, the discussion is in #111 . Please check it out @JamesHWade , @MichelNivard and provide feedback if possible 🙏

**Comment by JamesHWade** [comment]
@calderonsamuel, please take a look before I merge. There's a lot in here, and I welcome your edits and suggestions.

**Comment by calderonsamuel** [comment]
Sure! I'm having some free time on friday, and I see lots of changes! I'll try to have some benchmarks on the streaming part too cuz at first glance it seems a bit slower than before

**Comment by JamesHWade** [comment]
@calderonsamuel - Checking in to ask if you've had a chance to review the PR. I'm open to reverting changes these changes and going back to an R6 approach for later releases.

**Comment by calderonsamuel** [comment]
I apologize for keeping you waiting. I had anticipated having more free time, but it didn't turn out that way. I agree that it's preferable to merge these changes and then assess what can be later improved. Please just don't forget to add the changelog in the News file

**Comment by JamesHWade** [comment]
No need to apologize at all. I’ll update and the news and merge. Thank you for the guidance and suggestions building out these classes!

**Pull request merged**